types/react/index.d.ts

```diff
@@ -2732,16 +2732,18 @@ declare namespace React {
 // so boolean is only resolved for T = any
 type IsExactlyAny<T> = boolean extends (T extends never ? true : false) ? true : false;
 
+// Pick properties for which the value is `any`
+type PickAny<T> = Pick<T, { [K in keyof T]: IsExactlyAny<T[K]> extends true ? K : never }[keyof T]>;
```

Comment:
It looks like you're only using `PickAny` for the keys, right? Is the outer `Pick` needed then?
```ts
type ExactlyAnyPropertyKeys<T> = { [K in keyof T]: IsExactlyAny<T[K]> extends true ? K : never }[keyof T]
```